### PR TITLE
feat(ingestion): business-glossary - Add values and relatedTerms support

### PR DESCRIPTION
**Pull request by mohdsiddique**
* update GlossaryRelatedTerms.pdl 
* update source `datahub-business-glossary`
* update config `business_glossary.yml`

### CODE REVIEW

File: metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryRelatedTerms.pdl [hunk 1]
```diff
@@ -46,4 +46,43 @@ record GlossaryRelatedTerms {
...
+    }
+  }
+  @Searchable = {
+    "/*": {
+      "fieldName": "hasRelatedTermValues",
+      "fieldType": "URN",
+      "boostScore": 2.0
+    }
+  }
+  hasRelatedTermValues: optional array[GlossaryTermUrn]
```
Comment (gabe-lyons) [inline metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryRelatedTerms.pdl:67]: this should just be called `values`

File: metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryRelatedTerms.pdl [hunk 2]
```diff
@@ -46,4 +46,43 @@ record GlossaryRelatedTerms {
...
+    }
+  }
+  @Searchable = {
+    "/*": {
+      "fieldName": "isRelatedToTerms",
+      "fieldType": "URN",
+      "boostScore": 2.0
+    }
+  }
+  isRelatedToTerms: optional array[GlossaryTermUrn]
```
Comment (gabe-lyons) [inline metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryRelatedTerms.pdl:86]: call this `relatedTo`

File: metadata-ingestion/examples/bootstrap_data/business_glossary.yml [hunk 3]
```diff
@@ -59,3 +59,50 @@ nodes:
...
+        term_source: "EXTERNAL"
+        source_ref: FIBO
+        source_url: "https://spec.edmcouncil.org/fibo/ontology/FBC/ProductsAndServices/ClientsAndAccounts/Account"
+
+      - name: WindowColor
+        description: Supported window colors
+        term_source: "EXTERNAL"
+        source_ref: FIBO
+        source_url: "https://spec.edmcouncil.org/fibo/ontology/FBC/ProductsAndServices/ClientsAndAccounts/Account"
+        has_value:
```
Comment (gabe-lyons) [inline metadata-ingestion/examples/bootstrap_data/business_glossary.yml:91]: lets call this key `values`

File: metadata-ingestion/examples/bootstrap_data/business_glossary.yml [hunk 4]
```diff
@@ -59,3 +59,50 @@ nodes:
...
+        term_source: "EXTERNAL"
+        source_ref: FIBO
+        source_url: "https://spec.edmcouncil.org/fibo/ontology/FBC/ProductsAndServices/ClientsAndAccounts/Account"
+
+      - name: Spoon
+        description: an implement consisting of a small, shallow oval or round bowl on a long handle, used for eating, stirring, and serving food.
+        term_source: "EXTERNAL"
+        source_ref: FIBO
+        source_url: "https://spec.edmcouncil.org/fibo/ontology/FBC/ProductsAndServices/ClientsAndAccounts/Account"
+        is_related_to:
```
Comment (gabe-lyons) [inline metadata-ingestion/examples/bootstrap_data/business_glossary.yml:106]: call this key `related_to`

File: metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py [hunk 5]
```diff
@@ -41,6 +45,8 @@ class GlossaryTermConfig(ConfigModel):
     owners: Optional[Owners]
     inherits: Optional[List[str]]
     contains: Optional[List[str]]
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py:47]: let's add an additional Optional str field called `id`, which if present will be used as the id for the term. The id can be provided in `urn:li:..` form or in bare form.

File: metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py [hunk 6]
```diff
@@ -66,6 +72,8 @@ class DefaultConfig(ConfigModel):
 
 class BusinessGlossarySourceConfig(ConfigModel):
     file: str = Field(description="Path to business glossary file to ingest.")
+    enable_datahub_guid: bool = Field(
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py:75]: Suggest renaming this field to `enable_auto_id`  to line up with the suggested `id` field for terms and term groups.

File: metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py [hunk 7]
```diff
@@ -77,14 +85,22 @@ class BusinessGlossaryConfig(DefaultConfig):
     def version_must_be_1(cls, v):
         if v != "1":
             raise ValueError("Only version 1 is supported")
+        return v
+
+
+def create_id(path: List[str], enable_datahub_guid: bool) -> str:
```
Comment (shirshanka) [inline metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py:91]: this should additionally take in the Optional `id`.

File: metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryRelatedTerms.pdl (hunk 1 above)
Comment (shirshanka) [inline metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryRelatedTerms.pdl:67]: simplify to `values`

File: metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryRelatedTerms.pdl (hunk 2 above)
Comment (shirshanka) [reply metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryRelatedTerms.pdl:86]: or `relatedTerms`

File: metadata-ingestion/examples/bootstrap_data/business_glossary.yml (hunk 4 above)
Comment (shirshanka) [reply metadata-ingestion/examples/bootstrap_data/business_glossary.yml:106]: or just `related`

File: metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryRelatedTerms.pdl (hunk 2 above)
Comment (mohdsiddique) [reply metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryRelatedTerms.pdl]: Done

File: metadata-ingestion/examples/bootstrap_data/business_glossary.yml (hunk 3 above)
Comment (mohdsiddique) [reply metadata-ingestion/examples/bootstrap_data/business_glossary.yml]: Done

File: metadata-ingestion/examples/bootstrap_data/business_glossary.yml (hunk 4 above)
Comment (mohdsiddique) [reply metadata-ingestion/examples/bootstrap_data/business_glossary.yml]: Done

File: metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py (hunk 6 above)
Comment (mohdsiddique) [reply metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py]: Done

File: metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py (hunk 7 above)
Comment (mohdsiddique) [reply metadata-ingestion/src/datahub/ingestion/source/metadata/business_glossary.py]: Done

File: metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryRelatedTerms.pdl (hunk 1 above)
Comment (mohdsiddique) [reply metadata-models/src/main/pegasus/com/linkedin/glossary/GlossaryRelatedTerms.pdl]: Done

**Review by gabe-lyons — APPROVED** [review]

**Comment by gabe-lyons** [comment]
Cannot reproduce the datahub-ingestion docker issue locally- I believe it is spurious. Merging through.

**Pull request merged**